src/common/imageio.c

```diff
@@ -746,19 +746,48 @@ int dt_imageio_export_with_flags(const uint32_t imgid, const char *filename,
             ? FALSE
             : high_quality;
 
+  /* The pipeline might have out-of-bounds problems at the right and lower borders leading to
+     artefacts or mem access errors if ignored. (#3646)
+     It's very difficult to prepare the pipeline avoiding this **and** not introducing artefacts.
+     But we can test for that situation and if there is an out-of-bounds problem we
+     have basically two options:
+     a) reduce the output image size by one for width & height.
+     b) change the scale. In theory this marginally reduces quality.
+
+     These are the rules for export:
+     1. If we have the **full image** (defined by dt_image_t width, height and crops) we look for upscale.
+        If this is off use a), if on use b)
+     2. If we have defined format_params->max_width or/and height we use b)
+     3. Thumbnails are defined as in 2 so use b)
+     4. Cropped images are detected and use b)
+     5. Upscaled images use b)
+     6. Rotating by +-90° does not change the output size.
+     7. Never generate images larger than requested.         
+  */
+
+  const gboolean iscropped =
```

Comment:
Shouldn't this be:
```

  const gboolean iscropped =
    ((pipe.processed_width < img->crop_width) ||
     (pipe.processed_height < img->crop_height));
```
crop_x - crop_width looks suspicious to me.

Let me check on all integration tests.

Reply:
I confirm, with this change all the integration tests are passing fine, the export size is the same for all tests. So this looks good.